main/src/cgeo/geocaching/connector/trackable/GeokretyParser.java

```diff
@@ -324,4 +442,25 @@ protected static String getType(final int type) {
         }
         return null;
     }
+
+    /**
+     * Find spotted username if last move is Grab/Visit. (Skip comments)
+     *
+     * @param logsEntries
+     *          the log entries to analyze
+     * @return
+     *          The spotted username (or unkonwn)
+     */
+    private static String getLastSpottedUsername(final List<LogEntry> logsEntries) {
+        for (LogEntry log: logsEntries) {
+            if (log.getType() == LogType.NOTE) {
+                continue;
+            }
+            if (log.getType() == LogType.GRABBED_IT || log.getType() == LogType.VISIT) {
+                return log.author;
+            }
+            break;
```

Comment:
I'm not really sure what exactly you want to calculate here. If you just want to find any grabbed or visited log, then neither the continue nor the break should be part of the loop, but just the if.
If you want to calculate something else, you still should rework it, to not have a continue _and_ a break in the loop. For me, at most one of those is allowed in a loop to make it understandable.